tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/OpenTelemetry/StartSpanIntegration.cs

```diff
@@ -0,0 +1,79 @@
+        MinimumVersion = "1.0.0",
+        MaximumVersion = "1.0.0",
+        IntegrationName = nameof(Configuration.IntegrationId.OpenTelemetry))]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class StartSpanIntegration
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(StartSpanIntegration));
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTracer">Type of the tracer</typeparam>
+        /// <typeparam name="TSpanKind">Type of the span kind</typeparam>
+        /// <typeparam name="TSpanContext">Type of the span context</typeparam>
+        /// <typeparam name="TSpanAttributes">Type of the span attributes</typeparam>
+        /// <typeparam name="TLinks">Type of the span links</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="name">Span name.</param>
+        /// <param name="kind">Span kind.</param>
+        /// <param name="parentContext">Parent context for new span.</param>
+        /// <param name="spanAttributes">Initial attributes for the span.</param>
+        /// <param name="links">Links for the span.</param>
+        /// <param name="startTimeinstance">Start time for the span.</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTracer, TSpanKind, TSpanContext, TSpanAttributes, TLinks>(TTracer instance, string name, TSpanKind kind, ref TSpanContext parentContext, TSpanAttributes spanAttributes, TLinks links, DateTimeOffset startTimeinstance)
+        {
+            Log.Information("Entered OnMethodBegin for StartSpan");
+            return new CallTargetState(Tracer.Instance.InternalActiveScope);
+        }
+
+        /// <summary>
+        /// OnMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TReturn">Type of the return value</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="returnValue">Return value.</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>CallTargetReturn</returns>
+        internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
+        {
+            Log.Information("Entered OnMethodEnd for StartSpan");
+
+            // If the integration created a new scope, dispose it
+            var previousScope = state.Scope;
+            var currentScope = Tracer.Instance.InternalActiveScope;
+
+            if (previousScope != currentScope)
```

Comment:
Same here, I think we should check activity integration is enabled?

Reply:
Added in 5e6024b64f5cad7671d2a9256f0b60ebe69ea95c